packages/wallet/ui/src/service/Offers.js

```diff
@@ -1,36 +1,106 @@
+// @ts-check
+
 import {
   makeNotifierKit,
   makeAsyncIterableFromNotifier,
 } from '@agoric/notifier';
+import { E } from '@endo/eventual-send';
+
 import {
   loadOffers as load,
   removeOffer as remove,
   addOffer as add,
+  watchOffers,
 } from '../store/Offers.js';
 
+/** @typedef {import('@agoric/smart-wallet/src/types.js').Petname} Petname */
+
+export const OfferStatus = {
+  proposed: 'proposed',
+  accepted: 'accept',
+  rejected: 'rejected',
+  pending: 'pending',
+  declined: 'decline',
+};
+
 /**
- * @param {string} publicAddress
+ * @param {string} chainId
+ * @param {string} address
  * @param {(data: string) => Promise<any>} signSpendAction
  * @param {Notifier<any>} chainOffersNotifier
+ * @param {Marshaller} boardIdMarshaller
  */
 export const getOfferService = (
-  publicAddress,
+  chainId,
+  address,
   signSpendAction,
   chainOffersNotifier,
+  boardIdMarshaller,
 ) => {
   const offers = new Map();
-  let chainOffers = [];
   const { notifier, updater } = makeNotifierKit();
-  const broadcastUpdates = () =>
-    updater.updateState([...offers.values(), ...chainOffers]);
+  const broadcastUpdates = () => updater.updateState([...offers.values()]);
+
+  const unserializeAndAddSpendAction = async (pursePetnameToBrand, offer) => {
+    const {
+      id,
+      instanceHandleBoardId,
+      invitationMaker: { method },
+      proposalTemplate: { give, want },
+    } = offer;
+
+    const mapPursePetnamesToBrands = entries =>
+      Object.fromEntries(
+        Object.entries(entries).map(([kw, { pursePetname, value }]) => [
```

Comment:
the param must not be `entries` or we wouldn't need to call `entries()` on it. Please name the param by what data it conveys, not its shape, the use jsdoc type to describe its shape.